pkg/machine/fcos.go

```diff
@@ -107,37 +107,50 @@ type FcosDownload struct {
 	Download
 }
 
-func NewFcosDownloader(vmType VMType, vmName string, imageStream FCOSStream, vp VirtProvider) (DistributionDownload, error) {
-	info, err := GetFCOSDownload(vp, imageStream)
+func NewDownload(vmName string, vp VirtProvider) (Download, error) {
```

Comment:
Should this be a VirtProvider function?

func (vp *VirtProfider) NewDownload(vmName string) (Download, error)

Reply:
Also, should probably return *Download - so you don't have to return Download{} but can return nil instead on error